app/models/user.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_one :profile, dependent: :destroy
+
+  # User is post author
+  has_many :posts, foreign_key: :author_id, dependent: :destroy
+
+  # User is post receiver
+  has_many :posts, as: :postable, dependent: :destroy
```

Comment:
The idea is:

1. The User is a postable, meaning posts can be made on the User's page
2. The seeming repetition of the relationship between the User and Post is because on one of them the User is the author of the Post (which is line 12), the other one is where the User is the recipient of a post from another user (line 15)